diracx-db/src/diracx/db/sql/job/db.py

```diff
@@ -189,243 +238,178 @@ async def getJobJDL(self, job_id: int, original: bool = False) -> str:
+
+        if original:
+            stmt = select(JobJDLs.JobID, JobJDLs.OriginalJDL).where(
+                JobJDLs.JobID.in_(job_ids)
+            )
+        else:
+            stmt = select(JobJDLs.JobID, JobJDLs.JDL).where(JobJDLs.JobID.in_(job_ids))
+
+        return {
+            jobid: extractJDL(jdl)
+            for jobid, jdl in (await self.conn.execute(stmt))
+            if jdl
+        }
+
+    async def insert_bulk(
         self,
-        jdl,
-        owner,
-        owner_group,
-        initial_status,
-        initial_minor_status,
-        vo,
+        jobs: list[JobSubmissionSpec],
     ):
         from DIRAC.Core.Utilities.ClassAd.ClassAdLight import ClassAd
         from DIRAC.Core.Utilities.ReturnValues import returnValueOrRaise
         from DIRAC.WorkloadManagementSystem.DB.JobDBUtils import (
             checkAndAddOwner,
+            compressJDL,
             createJDLWithInitialStatus,
-            fixJDL,
         )
 
-        job_attrs = {
-            "LastUpdateTime": datetime.now(tz=timezone.utc),
-            "SubmissionTime": datetime.now(tz=timezone.utc),
-            "Owner": owner,
-            "OwnerGroup": owner_group,
-            "VO": vo,
-        }
-
-        jobManifest = returnValueOrRaise(checkAndAddOwner(jdl, owner, owner_group))
-
-        jdl = fixJDL(jdl)
-
-        job_id = await self._insertNewJDL(jdl)
-
-        jobManifest.setOption("JobID", job_id)
+        jobs_to_insert = []
+        jdls_to_update = []
+        inputdata_to_insert = []
+        original_jdls = []
 
-        job_attrs["JobID"] = job_id
-
-        # 2.- Check JDL and Prepare DIRAC JDL
-        jobJDL = jobManifest.dumpAsJDL()
-
-        # Replace the JobID placeholder if any
-        if jobJDL.find("%j") != -1:
-            jobJDL = jobJDL.replace("%j", str(job_id))
+        # generate the jobIDs first
+        for job in jobs:
+            original_jdl = deepcopy(job.jdl)
+            jobManifest = returnValueOrRaise(
+                checkAndAddOwner(original_jdl, job.owner, job.owner_group)
+            )
 
-        class_ad_job = ClassAd(jobJDL)
-        class_ad_req = ClassAd("[]")
-        if not class_ad_job.isOK():
-            job_attrs["Status"] = JobStatus.FAILED
+            # Fix possible lack of brackets
+            if original_jdl.strip()[0] != "[":
+                original_jdl = f"[{original_jdl}]"
 
-            job_attrs["MinorStatus"] = "Error in JDL syntax"
+            original_jdls.append((original_jdl, jobManifest))
 
-            await self._insertJob(job_attrs)
+        job_ids = await get_inserted_job_ids(
+            self.conn,
+            JobJDLs.__table__,
+            [
+                {
+                    "JDL": "",
+                    "JobRequirements": "",
+                    "OriginalJDL": compressJDL(original_jdl),
+                }
+                for original_jdl, _ in original_jdls
+            ],
+        )
```

Comment:
We need to do single row inserts for this to be safe when creating job IDs. use asyncio taskgroup to do it